### PR TITLE
1570 Update dataset creat, edit and export methods

### LINKED ISSUE #1570: Menas API for Dataset properties - pilot property Data Classification
## Background
Dataset properties is a map attached to a Dataset.
The keys of the map are defined on application setup level. They are common to all Datasets of the deployment.
The values then are entered by users and are specific per each Dataset.

## Feature
Implement the Menas API per the design, fulfulling the needs to serve the pilot Dataset property - _Data Classification_.
Also add the specific configuration

### Data Classification
* Selection from a given list
* Consider it Mandatory
* Consider writing it to _INFO file

### PR DESCRIPTION
**Pull request by Zejnilovic**
Not sure if in the scope of this PR to add validations on `create`, `replaceProperties` or `import`.

Thanks to the previous broad implementation of property definitions (#1569), this should be enough to start implementing the UI changes.

FIxes #1570

RN: Properties now can be added to dataset on create, edit or import. They are expected as an object of key-value pairs. Export function exports them as the same.

### CODE REVIEW

**Review by benedeki — APPROVED** [review]
- [x] code reviewed
- [ ] pulled
- [ ] built
- [ ] run

**Comment by Zejnilovic** [comment]
even after `mvn clean install -Pintegration` migration module does not build for me locally. Tests Fail. I would like for someone to run it locally as well to see the test pass not only on Jenkins.

**Review by AdrianOlosutean — APPROVED** [review]
Code reviewed and ran `mvn clean install -Pintegration`

File: menas/src/main/scala/za/co/absa/enceladus/menas/services/DatasetService.scala [hunk 1]
```diff
@@ -57,6 +57,7 @@ class DatasetService @Autowired()(datasetMongoRepository: DatasetMongoRepository
           .setHDFSPath(dataset.hdfsPath)
           .setHDFSPublishPath(dataset.hdfsPublishPath)
           .setConformance(dataset.conformance)
+          .setProperties(dataset.properties)
```
Comment (dk1844) [inline menas/src/main/scala/za/co/absa/enceladus/menas/services/DatasetService.scala:60]: Thanks for this fix, @Zejnilovic

File: menas/src/main/scala/za/co/absa/enceladus/menas/services/DatasetService.scala [hunk 2]
```diff
@@ -106,7 +107,8 @@ class DatasetService @Autowired()(datasetMongoRepository: DatasetMongoRepository
       hdfsPublishPath = newDataset.hdfsPublishPath,
       schemaName = newDataset.schemaName,
       schemaVersion = newDataset.schemaVersion,
-      conformance = List())
+      conformance = List(),
+      properties = newDataset.properties)
```
Comment (dk1844) [inline menas/src/main/scala/za/co/absa/enceladus/menas/services/DatasetService.scala:111]: dtto

**Review by dk1844 — APPROVED** [review]
(just read through)

**Comment by Zejnilovic** [comment]
I could really use that `test ran` label

**Review by HuvarVer — APPROVED** [review]
Basic checks
- [ ]  code reviewed
- [x] pulled
- [x] built
- [x] run

REST API test
- [x]  create Dataset
- [ ] edit Dataset
- [ ] export Dataset
- [ ] import Dataset
- [ ] replace Properties

UI tests
- [x]  create dataset
- [x] edit dataset
- [x] delete dataset

spark jobs test
- [x] run standardization
- [x] run conformance

**Pull request merged**